README.md

```diff
@@ -1,2 +1,131 @@
-# machine-learning-integration-plugin
-A CRS plugin to use machine learning together with the rule set.
+# OWASP ModSecurity Core Rule Set - Machine Learning Plugin
+
+## Description
+
+This is a plugin that serves to integrate machine learning support to the CRS.
+
+The plugin consists of 3 parts -
+1. Plugin rules
+2. Lua script
+3. Flask server
+
+### Plugin Rules
+The plugin contains rules which call the Lua script and pass or block the request based on the status returned to the rule.
+
+### Lua Script
+The lua script receives the request from the rule, reads the data sent in the request and reconstructs the request to form a http POST request to be sent to the flask server ```ml_model_server```. 
+After receiving the response from the server, the lua script sends the status response back to the CRS plugin. 
+
+### Flask Server
+This server receives a http POST request from the lua script. It extracts the different parameters from the request. The parameters extracted are request method, path, arguements, file names, file sizes, hour and day. These parameters are sent to a function which is supposed to call a machine learning model. This function has been stubbed by a random function for now. 
+This function is supposed to return a ml_anomaly_score. Based on this score, the server reuturns a 200 OK or 401 status. 
```

Comment:
Server also needs to return some kind of a score in the body of the response and without new line and so (just a score probably in a form of a integer), at least Lua script is expecting it. Should be noted in the documentation.